specs/ServerCertificate.md

```diff
@@ -0,0 +1,229 @@
+{
+  // Safeguarding the handler when unsupported runtime is used.
+  try
+  {
+    if (!_isServerCertificateError)
+    {
+      webView.CoreWebView2.ReceivingServerCertificateError += WebView_ReceivingServerCertificateError;
+    }
+    else
+    {
+      webView.CoreWebView2.ReceivingServerCertificateError -= WebView_ReceivingServerCertificateError;
+    }
+    _isServerCertificateError = !_isServerCertificateError;
+
+    MessageBox.Show(this,
+      _isServerCertificateError ? "Custom server certificate error has been enabled" : "Custom server certificate error has been disabled",
+      "Custom server certificate error");
+  }
+  catch (NotImplementedException exception)
+  {
+    MessageBox.Show(this, "Custom server certificate error Failed: " + exception.Message, "Custom server certificate error");
+  }
+}
+
+void WebView_ReceivingServerCertificateError(object sender, CoreWebView2ReceivingServerCertificateErrorEventArgs e)
+{
+  if (e.ErrorStatus == CoreWebView2WebErrorStatus.CertificateIsInvalid)
+  {
+    // Continue the request with the SSL certificate.
+    e.Handled = true;
+  }
+  else
+  {
+    // Cancel the request.
+    e.Cancel = true;
+  }
+}
+
+// This example clears the SSL decision in response to proceeding with SSL certificate errors.
+void ClearSSLDecision()
+{
+  webView.CoreWebView2.ClearSSLDecision();
+  MessageBox.Show(this, "Cleared", "Clear SSL Decision");
+}
+```
+
+# API Details
+
+## Win32 C++
+``` cpp
+[uuid(4B7FF0D2-8203-48B0-ACBF-ED9CFF82567A), object, pointer_default(unique)]
+interface ICoreWebView2_11 : ICoreWebView2_10 {
+  /// Add an event handler for the ServerCertificateError event.
+  /// The ServerCertificateError event is raised when the WebView2
+  /// cannot verify server's digital certificate while loading a web page.
+  ///
+  /// With this event you have several options for responding to server
+  /// certificate error requests:
+  ///
+  /// Scenario                                                   | Handled | Cancel
+  /// ---------------------------------------------------------- | ------- | ------
+  /// Ignore the warning and continue the request                | True    | False
+  /// Cancel the request                                         | n/a     | True
+  /// Display default SSL interstitial page                      | False   | False
+  ///
+  /// If you don't handle the event, WebView2 will show the default SSL interstitial page to user.
+  ///
+  /// \snippet SettingsComponent.cpp ServerCertificateError1
+  HRESULT add_ReceivingServerCertificateError(
+      [in] ICoreWebView2ReceivingServerCertificateErrorEventHandler* eventHandler,
+      [out] EventRegistrationToken * token);
+  /// Remove an event handler previously added with add_ReceivingServerCertificateError.
+  HRESULT remove_ReceivingServerCertificateError([in] EventRegistrationToken token);
+
+  /// Clears the SSL decision in response to proceeding with SSL certificate errors.
+  HRESULT ClearSSLDecision();
```

Comment:
How does this relate to the ClearBrowsingData method? Should this be an enum in that method rather than its own method?

Reply:
It does make sense to include in `ClearBrowsingData` method if we persist the decisions to profile. For webview2 we are storing it in memory (to be align with Android WebView& CEF) unlike browser. So, IMO providing a separate method would be better. Correct me if I am wrong.

Reply:
You mean this information is per-session unlike everything in ClearBrowsingData. Ok